File: Anul II/Semestrul II/Medii de proiectare si programare/AgentiiZboruriC/repository/PersoanaDbRepo.cs
using System.Collections;
using System.Data.SQLite;
using AgentiiZboruriC.domain;
namespace AgentiiZboruriC.repository;
using System.Diagnostics;
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using log4net;

public class PersoanaDbRepo : Repository<int, Persoana>
    {
        private readonly DbUtils dbUtils;

        public PersoanaDbRepo(DbUtils dbUtils)
        {
            this.dbUtils = dbUtils;
        }

        public Persoana? FindOne(object o)
        {
            using (SQLiteConnection con = dbUtils.GetConnection())
            using (SQLiteCommand cmd = new SQLiteCommand("SELECT * FROM Persoana WHERE id=@id", con))
            {
                cmd.Parameters.AddWithValue("@id", (int)o);
                using (SQLiteDataReader result = cmd.ExecuteReader())
                {
                    if (result.Read())
                    {
                        int id = result.GetInt32(0);
                        string nume = result.GetString(1);
                        string prenume = result.GetString(2);
                        return new Persoana(id, nume, prenume);
                    }
                }
            }
            return null;
        }

        public Zbor FindOne(int id)
        {
            throw new NotImplementedException();
        }

        public IEnumerable<Persoana> FindAll()
        {
            List<Persoana> persoane = new List<Persoana>();
            using (SQLiteConnection con = dbUtils.GetConnection())
            using (SQLiteCommand cmd = new SQLiteCommand("SELECT * FROM Persoana", con))
            {
                using (SQLiteDataReader result = cmd.ExecuteReader())
                {
                    while (result.Read())
                    {
                        int id = result.GetInt32(0);
                        string nume = result.GetString(1);
                        string prenume = result.GetString(2);
                        persoane.Add(new Persoana(id, nume, prenume));
                    }
                }
            }
            return persoane;
        }

        Optional<Persoana> Repository<int, Persoana>.Save(Persoana entity)
        {
            throw new NotImplementedException();
        }

        public Optional<Persoana> Delete(int id)
        {
            throw new NotImplementedException();
        }

        Optional<Persoana> Repository<int, Persoana>.Update(Persoana entity)
        {
            throw new NotImplementedException();
        }

        public Persoana? Save(Persoana entity)
        {
            using (SQLiteConnection con = dbUtils.GetConnection())
            using (SQLiteCommand cmd = new SQLiteCommand("INSERT INTO Persoana (id, nume, prenume) VALUES (@id, @nume, @prenume)", con))
            {
                cmd.Parameters.AddWithValue("@id", entity.Id);
                cmd.Parameters.AddWithValue("@nume", entity.Nume);
                cmd.Parameters.AddWithValue("@prenume", entity.Prenume);
                cmd.ExecuteNonQuery();
            }
            return null;
        }

        public Persoana? Delete(object o)
        {
            return null;
        }

        public Persoana? Update(Persoana entity)
        {
            return null;
        }

        public IEnumerable<Persoana> ChangeEntities(Dictionary<int, Persoana> entities)
        {
            return null;
        }
    }

File: Anul II/Semestrul II/Medii de proiectare si programare/AgentiiZboruri/Persistence/src/main/java/ro/persistence/PersoanaDbRepo.java
package ro.persistence;

import ro.model.Entity;
import ro.model.Persoana;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;
import java.util.Optional;

public class PersoanaDbRepo implements Repository{
    private final DbUtils dbUtils;

    public PersoanaDbRepo(DbUtils dbUtils) {
        this.dbUtils = dbUtils;
    }

    @Override
    public Optional findOne(Object o) {
        Connection con = dbUtils.getConnection();
        try (PreparedStatement preStmt = con.prepareStatement("SELECT * FROM Persoana WHERE id=?")) {
            preStmt.setInt(1, (Integer) o);
            try (ResultSet result = preStmt.executeQuery()) {
                if (result.next()) {
                    Integer id=result.getInt("id");
                    String nume=result.getString("nume");
                    String prenume=result.getString("prenume");
                    Persoana persoana = new Persoana(id,nume,prenume);
                    return Optional.of(persoana);
                }
            }
        } catch (SQLException ex) {
            System.out.println("Error DB " + ex);
        }
        return Optional.empty();
    }

    @Override
    public Iterable findAll() {
        Connection con = dbUtils.getConnection();
        List<Persoana> persoane = new ArrayList<>();
        try (PreparedStatement preStmt = con.prepareStatement("SELECT * FROM Persoana")) {
            try (ResultSet result = preStmt.executeQuery()) {
                while (result.next()) {
                    Integer id=result.getInt("id");
                    String nume=result.getString("nume");
                    String prenume=result.getString("prenume");
                    Persoana persoana = new Persoana(id,nume,prenume);
                    persoane.add(persoana);
                }
            }
        } catch (SQLException ex) {
            System.out.println("Error DB " + ex);
        }
        return persoane;
    }

    @Override
    public Optional save(Entity entity) {
        Connection con = dbUtils.getConnection();
        try (PreparedStatement preStmt = con.prepareStatement("INSERT INTO Persoana (id, nume, prenume) VALUES (?, ?, ?)")) {
            Persoana persoana = (Persoana) entity;
            preStmt.setInt(1, persoana.getId());
            preStmt.setString(2, persoana.getNume());
            preStmt.setString(3, persoana.getPrenume());
            preStmt.executeUpdate();
        } catch (SQLException ex) {
            System.out.println("Error DB " + ex);
            return Optional.of(entity);
        }
        return Optional.empty();
    }

    @Override
    public Optional delete(Object o) {
        return Optional.empty();
    }

    @Override
    public Optional update(Entity entity) {
        return Optional.empty();
    }

    @Override
    public Iterable changeEntities(Map entities) {
        return null;
    }
}